### DISCUSSION TITLE
ENH: to_wkt and to_string should use same rounding_precision (non-blocking surprise behavior)

### DISCUSSION THREAD

**Discussion by ludwick** [geopandas/geopandas#3350] [Q&A] [answered]
#### Is your feature request related to a problem?

The GeoSeries `to_wkt` and `to_string` seem to use precision 6 and 5 respectively when rounding polygon values.  I discovered this when migrating some code using `to_string` to get WKT strings to use `to_wkt` (so the global pandas option `display.max_colwidth` didn't have to be set unnessarily).

#### Describe the solution you'd like

By default `to_wkt` and `to_string` seem like they should use the same precision.

#### API breaking implications

Maybe some depend on this difference? But it seems unlikely.

#### Describe alternatives you've considered

Easily worked around by specifying precision explicitly when using `to_wkt`.

#### Additional context

Reproduction

```python
#!/usr/bin/env python
# Assumed versions:
# python 3.10.14
# pandas 2.1.4
# geopandas 0.13.2
# shapely 2.0.2

import pandas as pd
import geopandas as gpd
from shapely.geometry import shape

if __name__ == "__main__":
    poly = {'type': 'MultiPolygon', 'coordinates': [[[[-90.74692052687222, 34.14965959326743], [-90.74645839792734, 34.1496650154515], [-90.74639132715382, 34.14921675664262], [-90.74582873372799, 34.14926691254273], [-90.74535241853518, 34.149474895637155], [-90.74496863902432, 34.14956800829312], [-90.74419492047997, 34.14941223793254], [-90.74398879146895, 34.14987419107862], [-90.74117809972682, 34.15004823231045], [-90.74100974893992, 34.15002175244208], [-90.74087509933273, 34.14936830343046], [-90.74052843932809, 34.14912546854216], [-90.74068529011195, 34.14796014778624], [-90.74076064613143, 34.14728584165832], [-90.74091556665178, 34.14703776936295], [-90.7408923211989, 34.14640685260206], [-90.74100876239714, 34.14619641760211], [-90.7461818311732, 34.14710268892608], [-90.74648577147886, 34.1469498377466], [-90.74678644351951, 34.14707277543532], [-90.74688434605332, 34.14730865132611], [-90.74692052687222, 34.14965959326743]]]]}
    geometry = shape(poly)
    geometry = geometry.simplify(0.00015, preserve_topology=True)

    gdf = gpd.GeoDataFrame(geometry=[geometry], columns=["geometry"], crs="EPSG:4326")
    hull = gdf["geometry"].convex_hull

    to_wkt_val = hull.geometry.to_wkt()[0]

    pd.set_option("display.max_colwidth", None)
    to_string_val = hull.geometry.to_string(index=False, header=False)

    to_wkt_val_5 = hull.geometry.to_wkt(rounding_precision=5)[0]
    print(f"Shape: {poly}")
    print("to_wkt_val:")
    print(to_wkt_val)
    print("\nto_string_val:")
    print(to_string_val)
    print("\nto_wkt_val_5 (rounding_precision 5):")
    print(to_wkt_val_5)
```

I haven't dug into the source yet but if it's easy I may create a PR for this.

**Comment by martinfleis** [comment, marked as answer]
We define this within https://github.com/geopandas/geopandas/blob/0618b227086ef605c476b4a36449d5f65b9d859f/geopandas/array.py#L1569

and the situation is slightly different. We return 5 decimals for geographic CRS and 3 for projected CRS. The intention behind `_formatter` is to provide an output useful within the `print` method, not necessarily to provide a dtype conversion method. pandas directly says in the [documentation](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_string.html) "_Render a DataFrame to a console-friendly tabular output._", which is what we follow here.

I don't think this behaviour should change. The precision logic from `_formatter` works nicely in most cases to _see_ the data and if you want to convert your data to WKT, you should explicitly use `to_wkt()` method. It is also worth noting that `to_wkt` will be significantly faster.

**Comment by ludwick** [comment]
Ahh! Thank you for explaining why they are different (and now if someone searches issues they'll find this). I was converting code to what seemed more reasonable to me – the use of to_wkt – and also glad to learn it's a faster/better way to do this! Feel free to close this.

**Comment by ludwick** [comment]
I meant to add: unfortunately the examples I found around the internet seem to encourage the use of the use of the pandas option plus use of to_string which is presumably partly why this code (that I've converted) was written this way. Anyway thank you!
